clients/cli/src/prover_runtime.rs

```diff
@@ -0,0 +1,446 @@
+//! Prover Runtime
+//!
+//! Handles background execution of proof tasks in both authenticated and anonymous modes.
+//! Spawns async workers, dispatches tasks, and reports progress back to the UI.
+//!
+//! Includes:
+//! - Task fetching from the orchestrator (authenticated mode)
+//! - Worker management and task dispatching
+//! - Prover event reporting
+
+use crate::orchestrator::error::OrchestratorError;
+use crate::orchestrator::{Orchestrator, OrchestratorClient};
+use crate::prover::authenticated_proving;
+use crate::task::Task;
+use chrono::Local;
+use ed25519_dalek::{SigningKey, VerifyingKey};
+use nexus_sdk::stwo::seq::Proof;
+use sha3::{Digest, Keccak256};
+use std::time::Duration;
+use tokio::sync::{broadcast, mpsc};
+use tokio::task::JoinHandle;
+
+/// Events emitted by prover (worker) threads.
+#[allow(unused)]
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum WorkerEvent {
+    TaskFetcher { data: String },
+    Prover { worker_id: usize, data: String },
+    ProofSubmitter { data: String },
+}
+
+/// Starts authenticated workers that fetch tasks from the orchestrator and process them.
+pub async fn start_authenticated_workers(
+    node_id: u64,
+    signing_key: SigningKey,
+    orchestrator: OrchestratorClient,
+    num_workers: usize,
+    shutdown: broadcast::Receiver<()>,
+) -> (mpsc::Receiver<WorkerEvent>, Vec<JoinHandle<()>>) {
+    let mut join_handles = Vec::new();
+    // Worker events
+    let (event_sender, event_receiver) = mpsc::channel::<WorkerEvent>(100);
+
+    // Task fetching
+    let task_queue_size = 100;
+    let (task_sender, task_receiver) = mpsc::channel::<Task>(task_queue_size);
+    let verifying_key = signing_key.verifying_key();
+    let fetch_prover_tasks_handle = {
+        let orchestrator = orchestrator.clone();
+        let event_sender = event_sender.clone();
+        let shutdown = shutdown.resubscribe(); // Clone the receiver for task fetching
+        tokio::spawn(async move {
+            fetch_prover_tasks(
+                node_id,
+                verifying_key,
+                Box::new(orchestrator),
+                task_sender,
+                event_sender,
+                shutdown,
+            )
+            .await;
+        })
+    };
+    join_handles.push(fetch_prover_tasks_handle);
+
+    // Workers
+    let (result_sender, result_receiver) = mpsc::channel::<(Task, Proof)>(1000);
+
+    let (worker_senders, worker_handles) = start_workers(
+        num_workers,
+        result_sender.clone(),
+        event_sender.clone(),
+        shutdown.resubscribe(),
+    );
+    join_handles.extend(worker_handles);
+
+    // Dispatch tasks to workers
+    let dispatcher_handle = start_dispatcher(task_receiver, worker_senders, shutdown.resubscribe());
+    join_handles.push(dispatcher_handle);
+
+    // Send proofs to the orchestrator
```

Comment:
What's the reason for disentangling proving from sending of proofs? Just so that network conditions don't block local operations?